### ISSUE TITLE
Enhance Multivariate MODE to read input data only once rather than multiple times

### ISSUE THREAD

**Issue by davidalbo** [dtcenter/MET#2707]
*Replace italics below with details for this issue.*

## Describe the Enhancement ##
Current implementation reads data twice ('part1 simple' and 'part1 merge'.  Store in memory instead as part of part1 simple, access that in part1 merge. 

### Time Estimate ###
24 hours

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
January 16, 2024  (beta3)

### Funding Source ###
NOAA base

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Review default **alert** labels
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Milestone and Projects ###
- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**
- [ ] For the next official version, select the **MET-X.Y.Z Development** project

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ]  [MET](https://github.com/dtcenter/MET/issues/new/choose), 

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the next official version
Select: **MET-X.Y.Z Development** project for development toward the next official release
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by davidalbo** [comment]
This issue was resolved in the changes made here:   [https://github.com/dtcenter/MET/pull/2779](url). The new design reads in each input only once and keeps it in memory for use in several processing steps.
